File: app/schemas/supplier.py
"""Schémas Pydantic pour les fournisseurs."""
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr


class SupplierCreate(BaseModel):
    name: str
    contact_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    notes: Optional[str] = None


class SupplierUpdate(BaseModel):
    name: Optional[str] = None
    contact_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    notes: Optional[str] = None


class SupplierRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    tenant_id: int
    name: str
    contact_name: Optional[str]
    email: Optional[str]
    phone: Optional[str]
    address: Optional[str]
    notes: Optional[str]
    is_active: bool

File: frontend/apps/marveline/src/types/supplier.ts
export interface Supplier {
  id: number
  tenant_id: number
  name: string
  contact_name: string | null
  email: string | null
  phone: string | null
  address: string | null
  notes: string | null
  is_active: boolean
}

export interface SupplierCreate {
  name: string
  contact_name?: string
  email?: string
  phone?: string
  address?: string
  notes?: string
}

export interface SupplierUpdate {
  name?: string
  contact_name?: string
  email?: string
  phone?: string
  address?: string
  notes?: string
}
